packages/next-auth/src/next/index.ts

```diff
@@ -20,12 +20,14 @@ async function NextAuthNextHandler(
 ) {
   const { nextauth, ...query } = req.query
 
+  options.canonical_url = options.canonical_url ?? req.headers["x-forwarded-host"]
+
   options.secret =
     options.secret ?? options.jwt?.secret ?? process.env.NEXTAUTH_SECRET
 
   const handler = await NextAuthHandler({
     req: {
-      host: detectHost(req.headers["x-forwarded-host"]),
```

Comment:
Why not use the `'x-forwarded-host'` header? We have that already set in place. If you're using **Vercel** I think it'll fit your use case. Did you try it?